tests/test_file_download.py

```diff
@@ -112,6 +112,21 @@ def test_disk_usage_warning(self, disk_usage_mock: Mock) -> None:
             _check_disk_space(expected_size=self.expected_size, target_dir=disk_usage_mock)
             assert len(w) == 0
 
+    def test_disk_usage_warning_with_non_existent_path(self) -> None:
+        # Test for not existent path
+        with warnings.catch_warnings(record=True) as w:
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter("always")
+            _check_disk_space(expected_size=self.expected_size, target_dir="not_existent_path")
+            assert len(w) == 0
+
+        # Test for relative path
+        with warnings.catch_warnings(record=True) as w:
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter("always")
+            _check_disk_space(expected_size=self.expected_size, target_dir="./not_existent_path")
+            assert len(w) == 0
```

Comment:
Yes that's exactly how I was seeing it except that 1 path must be relative (let's say `"path/to/not_existent_path"`) and the other one absolute (`"/path/to/not_existent_path"`) to check that both of them doesn't raise an exception.

Reply:
```suggestion
    def test_disk_usage_warning_with_non_existent_path(self) -> None:
        # Test for not existent (absolute) path
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            _check_disk_space(expected_size=self.expected_size, target_dir="path/to/not_existent_path")
            assert len(w) == 0

        # Test for not existent (relative) path
        with warnings.catch_warnings(record=True) as w:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            _check_disk_space(expected_size=self.expected_size, target_dir="/path/to/not_existent_path")
            assert len(w) == 0
```